lnrpc/switchrpc/switch_server.go

```diff
@@ -364,6 +370,188 @@ func (s *Server) findEligibleChannelID(pubKey *btcec.PublicKey,
 			amount)
 }
 
+// TrackOnion provides callers the means to query whether or not a payment
+// dispatched via SendOnion succeeded or failed.
+func (s *Server) TrackOnion(ctx context.Context,
+	req *TrackOnionRequest) (*TrackOnionResponse, error) {
+
+	hash, err := lntypes.MakeHash(req.PaymentHash)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"invalid payment_hash=%x: %v", req.PaymentHash, err)
+	}
+
+	if req.SharedSecrets != nil {
+		return nil, status.Errorf(codes.Unimplemented,
+			"unable to process shared secrets")
+	}
+
+	// NOTE(calvin): In order to decrypt errors server side we require
+	// either the combination of session key and hop public keys from which
+	// we can construct the shared secrets used to build the onion or,
+	// alternatively, the caller can provide the list of shared secrets used
+	// during onion construction directly if they wish to maintain route
+	// privacy from the server.
+	//
+	// TODO: If we want to support "oblivious sends", then we'll need to
+	// pass the shared secrets to the sphinx library.
+	log.Debugf("Looking up status of onion attempt_id=%d for payment=%v",
+		req.AttemptId, hash)
+
+	// Attempt to build the error decryptor with the provided session key
+	// and hop public keys.
+	errorDecryptor, err := buildErrorDecryptor(
+		req.SessionKey, req.HopPubkeys,
+	)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument,
+			"error building decryptor: %v", err)
+	}
+
+	if errorDecryptor == nil {
+		log.Debug("Unable to build error decrypter with information " +
+			"provided. Will defer error handling to caller")
+	}
+
+	// Query the switch for the result of the payment attempt via onion.
+	resultChan, err := s.cfg.HtlcDispatcher.GetAttemptResult(
+		req.AttemptId, hash, errorDecryptor,
+	)
+	if err != nil {
+		message, code := TranslateErrorForRPC(err)
+
+		log.Errorf("GetAttemptResult failed for attempt_id=%d of "+
+			" payment=%v: %v", hash, message)
+
+		// If this is a htlcswitch.ErrPaymentIDNotFound error, we need
+		// to transmit that fact to the client so they can know with
```

Comment:
comment is talking about a case that doesnt seem to be handled at this point? 
`If this is a htlcswitch.ErrPaymentIDNotFound err...` makes it seem like this should be followed by an `if errors.Is(...)` block

Reply:
comment still there just fyi :)